cfrontend/PrintClight.ml

```diff
@@ -79,9 +79,9 @@ let rec expr p (prec, e) =
   | Econst_int(n, _) ->
       fprintf p "%ld" (camlint_of_coqint n)
   | Econst_float(f, _) ->
-      fprintf p "%.15F" (camlfloat_of_coqfloat f)
+     fprintf p "%.18g" (camlfloat_of_coqfloat f)
   | Econst_single(f, _) ->
-      fprintf p "%.15Ff" (camlfloat_of_coqfloat32 f)
+     fprintf p "%.18g" (camlfloat_of_coqfloat32 f)
```

Comment:
Here you must leave a `f` or `F` suffix to indicate `float` type instead of the default `double` type.